packages/platforms/browser/lib/auto-instrumentation/network-request-plugin.ts

```diff
@@ -38,12 +43,17 @@ export class NetworkRequestPlugin implements Plugin<BrowserConfiguration> {
     }
   }
 
-  private trackRequest = (startContext: RequestStartContext): RequestEndCallback | undefined => {
+  private trackRequest: RequestStartCallback = (startContext) => {
     if (!this.shouldTrackRequest(startContext)) return
 
     const networkRequestInfo = this.networkRequestCallback({ url: startContext.url, type: startContext.type })
 
-    if (!networkRequestInfo) return
+    if (!networkRequestInfo) {
+      return {
+        onRequestEnd: undefined,
+        extraRequestHeaders: this.getExtraRequestHeaders(startContext)
```

Comment:
where not tracking the request / creating a network span the request headers will be determined from the context